wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -389,6 +390,58 @@ abstract class Wallet
     }
   }
 
+  def sendFromOutPoints(
+      outPoints: Vector[TransactionOutPoint],
+      address: BitcoinAddress,
+      feeRate: FeeUnit)(implicit ec: ExecutionContext): Future[Transaction] = {
+    require(
+      address.networkParameters.isSameNetworkBytes(networkParameters),
+      s"Cannot send to address on other network, got ${address.networkParameters}"
+    )
+    logger.info(s"Sending to $address at feerate $feeRate")
+    for {
+      utxoDbs <- spendingInfoDAO.findByOutPoints(outPoints)
+      diff = utxoDbs.map(_.outPoint).diff(outPoints)
+      _ = require(diff.isEmpty,
+                  s"Not all OutPoints belong to this wallet, diff $diff")
+      spentUtxos =
+        utxoDbs.filterNot(utxo => TxoState.receivedStates.contains(utxo.state))
+      _ = require(
+        spentUtxos.isEmpty,
+        s"Some out points given have already been spent, ${spentUtxos.map(_.outPoint)}")
+
+      prevTxFs = utxoDbs.map(utxo =>
+        transactionDAO.findByOutPoint(utxo.outPoint).map(_.get.transaction))
```

Comment:
Why not just return the tuple here so we don't have to worry about zipping incorrectly below? 

I.e. 

```scala
utxoTxTuple <- Future.sequence { 
    utxoDbs.map(utxo =>
        transactionDAO.findByOutPoint(utxo.outPoint)
        .map(txDb => (utxo, txDb.get.transaction))
}
```